src/webgpu/api/validation/encoding/queries/begin_end.spec.ts

```diff
@@ -56,7 +77,23 @@ Tests the invalid nesting of begin/end occlusion queries:
         { calls: [0, 1, 'end', 'end'] },
       ] as const
   )
-  .unimplemented();
+  .fn(async t => {
+    const querySet = createQuerySetWithType(t, 'occlusion', 2);
+
+    const encoder = createRenderEncoderWithQuerySet(t, querySet);
+
+    t.params.calls.forEach(i => {
+      if (i !== 'end') {
+        encoder.encoder.beginOcclusionQuery(Number(i));
```

Comment:
Typechecker should allow simply:
```suggestion
        encoder.encoder.beginOcclusionQuery(i);
```